pennylane/devices/qubit/sampling.py

```diff
@@ -217,6 +225,12 @@ def measure_with_samples(
         res for _, res in sorted(list(enumerate(all_res)), key=lambda r: flat_indices[r[0]])
     )
 
+    # append MCM samples
+    if mid_measurements:
+        sorted_res = list(sorted_res)
+        sorted_res.extend(list(mid_measurements.values()))
+        sorted_res = tuple(sorted_res)
```

Comment:
```suggestion
        sorted_res += tuple(mid_measurements.values())
```
Fewer type conversions.